Bonzo Mask Durability Cooldown

Vixid1 opened the pull request:
Adds durability cooldown to both fragged and non-fragged bonzo masks. Changed a bit of code for changing item's durability for less duplicate code.

![image](https://user-images.githubusercontent.com/52578495/210094267-bcb25aee-4d09-4a89-bdc4-e7cf21a6ce9a.png)

efefury commented:
Could you add it for Spirit Mask as well? 
This is the message: §r§6Second Wind Activated§r§a! §r§aYour Spirit Mask saved your life!§r

Vixid1 commented:
Added Spirit Mask support. Did some more refactoring of code, moved away from deprecated `getInternalNameForItem(ItemStack stack)` and replaced with `ItemResolutionQuery`, moved common code into functions reducing duplicate code more so than before.

![image](https://user-images.githubusercontent.com/52578495/210108412-44623086-d427-4be7-b02d-2c9b2d81836a.png)

romangraef commented on src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/ItemCooldowns.java:
```diff
@@ -153,36 +178,78 @@ private static boolean isPickaxe(String internalname) {
+
+			if (internalname != null) {
+				switch (item) {
+					case "Pickaxe":
```
Not a big fan of using magic strings like this. Maybe use an enum, or extract the strings into constants. Other than that great PR

Vixid1 replied:
I thought strings wasn't quite the correct thing at the time, and forgot about enums until you commented this so I've changed them to enums now.

jani270 reviewed (changes requested):
"Fixed Bonzo Mask and upgraded Bonzo Mask having separate cooldowns allowing for double the ability usage" 
[https://hypixel.net/threads/january-10th-some-bug-fixes.5233523/](https://hypixel.net/threads/january-10th-some-bug-fixes.5233523/)
